_posts/2017-07-23-fastlane-travis-weekly-deploys.md

```diff
@@ -0,0 +1,314 @@
+
+```ruby
+git_url "https://#{ENV['GITHUB_SUBMODULES_USER']}@github.com/artsy/mobile_fastlane_match"
+
+# Instead of 
+# git_url "https://github.com/artsy/mobile_fastlane_match"
+```
+
+This is one of the highlights on fastlane's choice in building a DSL that which sits above a real programming language, you give users a lot of flexibility.
+
+Next up, I added a fastlane lane for code signing, and keychain setup. This just calls two setup functions.
+
+```ruby
+lane :setup_signing do
+  create_keychain(
+    name: ENV['MATCH_KEYCHAIN_NAME'],
+    password: ENV['MATCH_PASSWORD'],
+    default_keychain: true,
+    unlock: true,
+    timeout: 3600,
+    add_to_search_list: true
+  )
+
+  match(type: 'appstore',
+        keychain_name: ENV['MATCH_KEYCHAIN_NAME'],
+        keychain_password: ENV['MATCH_PASSWORD'],
+        readonly: true)
+end
+```
+
+## Creating the build and shipping it to Testflight
+
+This is handled by [fastlane gym][gym] at the start of the main lane.
+
+```ruby
+# You can run `bundle exec fastlane ship` to test the build process locally.
+lane :ship do
+  # Run the code signing lane
+  setup_signing
+
+  # We were having issues with building an a few folders deep.
+  # The /Pods bit is because we can rely on it being there, see
+  # this link: https://docs.fastlane.tools/advanced/#directory-behavior
+  Dir.chdir('../emission/Example/Pods') do
+    gym workspace: 'Emission.xcworkspace',
+        configuration: 'Deploy',
+        scheme: 'Emission'
+  end
+
+  # [...]
+end
+```
+
+It uses a scheme for deploys, which prioritises using AppHub over a local React Native server. Gym handles a lot of CLI ugliness for us, and works well.
+
+Sending the app to Testflight involves a little bit of work:
+
+```ruby
+# Get the last 10 lines of the CHANGELOG for Testflight
+changelog = '../emission/CHANGELOG.md'
+upcoming_release_notes = File.read(changelog).split("\n### ").first
+
+# Ship to testflight
+pilot changelog: upcoming_release_notes
+```
+
+This lets the deploy process figure out what the latest release version is, and how many builds have shipped for that version. Then those can be used to set the build version and create a tag associated with it.
+
+[fastlane pilot][] is used to send off the compiled build to Testflight.
+
+## Keeping track of deploys
+
+I don't know when we'll need it today, but it's always good to be able to go back and see what code lines up to every release. To do this I have a few lines of Ruby that creates a tag inside the original Emission repo.
+
+```ruby
+# Do a tag, we use a http git remote so we can have push access
+# as the default remote for travis is read-only. This needs to be
+# inside the emission repo, instead of our own.
+Dir.chdir('../emission/Example/') do
+  tag = "deploy-#{latest_version}-#{build_version}"
```

Comment:
You'd like the [add_git_tag](https://docs.fastlane.tools/actions/#add_git_tag) and the [push_git_tags](https://docs.fastlane.tools/actions/#push_git_tags) actions